parallel-consumer-core/src/test-integration/java/io/confluent/parallelconsumer/integrationTests/state/PartitionStateCommittedOffsetIT.java

```diff
@@ -345,34 +379,21 @@ private List<PollContext<String, String>> runPcUntilOffset(OffsetResetStrategy o
 
             getKcu().produceMessages(getTopic(), 1, "poll-bumper");
 
-            if (offsetResetPolicy.equals(OffsetResetStrategy.NONE)) {
-                Awaitility.await().untilAsserted(() -> {
-                    assertWithMessage("PC crashed / failed fast").that(tempPc.isClosedOrFailed()).isTrue();
-                    assertThat(tempPc.getFailureCause()).hasCauseThat().hasMessageThat().contains("Error in BrokerPollSystem system");
-                    var stackTrace = ExceptionUtils.getStackTrace(tempPc.getFailureCause());
-                    Truth.assertThat(stackTrace).contains("Undefined offset with no reset policy for partitions");
-                });
-                return UniLists.of();
-            } else {
-
-                Awaitility.await()
-                        .failFast(tempPc::isClosedOrFailed)
-                        .untilAsserted(() -> {
-                            assertThat(seenOffsets).isNotEmpty();
-                            assertThat(seenOffsets.last().offset()).isGreaterThan(expectedProcessToOffset - 2);
-                        });
-
-                if (!succeededOffsets.isEmpty()) {
-                    log.debug("Succeeded up to: {}", succeededOffsets.last().offset());
-                }
-                log.debug("Consumed up to {}", seenOffsets.last().offset());
-
-                var sorted = new ArrayList<>(seenOffsets);
-                Collections.sort(sorted, Comparator.comparingLong(PollContext::offset));
+            Awaitility.await()
+                    .failFast(tempPc::isClosedOrFailed)
+                    .untilAsserted(() -> {
+                        assertThat(seenOffsets).isNotEmpty();
+                        assertThat(seenOffsets.last().offset()).isGreaterThan(expectedProcessToOffset - 2);
+                    });
 
-
-                return sorted;
+            if (!succeededOffsets.isEmpty()) {
+                log.debug("Succeeded up to: {}", succeededOffsets.last().offset());
             }
+            log.debug("Consumed up to {}", seenOffsets.last().offset());
+
+            var sorted = new ArrayList<>(seenOffsets);
+            Collections.sort(sorted, Comparator.comparingLong(PollContext::offset));
+            return sorted;
         } finally {
             try {
                 tempPc.close(); // close manually in this branch only, as in other branch it crashes
```

Comment:
not sure what that comment supposed to mean - leftover from branching logic ?

Reply:
any reason you are suppressing exception thrown by close()? 
is it possible for PC to fail close with exception when you are not expecting it or is it to prevent exception on close() overshadowing exception during execution (due to NONE offset reset policy / offset not present on poll)?  
Just thinking - if you only want to execute close when PC is not already closed or failed - cant you do same check as above on failFast i.e.
```
finally {
      if(!tempPc.isClosedOrFailed()) {
              temPc.close();
      }
}
```